homeassistant/helpers/__init__.py

```diff
@@ -98,14 +98,17 @@ def config_per_platform(config, domain, logger):
 
     while config_key in config:
         platform_config = config[config_key]
```

Comment:
This will raise a `KeyError` if `config_key` does not exist within `config`. The check `if playform_type is None` from inside the for loop should be checked before this line
```python
if config_key not in config:
  logger.warning('No platform specified for %s', config_key)
  break
```